src/ucp/api/ucp.h

```diff
@@ -2413,6 +2503,20 @@ ucs_status_ptr_t ucp_ep_flush_nb(ucp_ep_h ep, unsigned flags,
 ucs_status_ptr_t ucp_ep_flush_nbx(ucp_ep_h ep, const ucp_request_param_t *param);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @brief Query attributes of a specific endpoint.
+ *
+ * This routine fetches information about the endpoint.
+ * 
+ * @param [in]    ep   Endpoint to query.
+ * @param [inout] attr Filled with attributes of the endpoint.
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_ep_perf_query(ucp_ep_h ep, ucp_ep_perf_attr_t *attr);
```

Comment:
maybe use `ucp_ep_query` name to be able to extend this in the future? Otherwise we will have to introduce new call if want to query non-perf attrs

Reply:
I agree. we already made some plans to query an ep for the client-server case to retrieve an ep's local connected sockaddr and the remote sockaddr it's connected to.
if we dont use ucp_ep_query now, we would need to have different functions to querying different ep attrs.

Reply:
There is a difference between `ucp_ep_query` and `ucp_ep_perf_quer`y: in perf_query, the user would pass input parameters (memory types, length, operation) for which the performance should be retrieved. 
IMO it would be weird to mix those input parameters along with general ep query, and better have 2 separate functions.

Reply:
>  the user would pass input parameters (memory types, length, operation) for which the performance should be retrieved.

maybe use struct to combine them?

Reply:
> maybe use struct to combine them?
That struct will also need to have field mask

> then maybe avoid using 'query' in the name here?
agree it could be better, unfortunately I can't thing of a good name, maybe something like `ucp_ep_estimate_perf` or `ucp_ep_eval_perf` ?

Reply:
> > maybe use struct to combine them?
> 
> That struct will also need to have field mask
> 
> > then maybe avoid using 'query' in the name here?
> 
> agree it would be better, unfortunately I can't thing of a good name, maybe something like `ucp_ep_estimate_perf` or `ucp_ep_eval_perf` ?

`ucp_ep_calc_perf`?
`ucp_ep_eval_perf` sounds good to me.

Reply:
@gleon99  WDYT about `ucp_ep_eval_perf`?
i think we should sync the UCT API with same name as we decide here

Reply:
Sounds good. Maybe `ucp_ep_estimate_perf`? "estimation" is perhaps more accurate here, as the function **estimates** performance and returns an **estimate**, rather than **evaluating** it. But I'm ok with `_eval_` as well.

Reply:
ok, let's use estimate

Reply:
@yosefe let's discuss offline. This does look like a query interface and I'm still not sure why want separate perf API. I do think it is the right functionality to have we just need to flush out the API